hydra/core/utils.py

```diff
@@ -11,18 +11,28 @@
 from os.path import splitext
 from pathlib import Path
 from textwrap import dedent
-from typing import Any, Dict, Optional, Sequence, Union, cast
+from typing import TYPE_CHECKING, Any, Dict, Optional, Sequence, Union, cast
 
 from omegaconf import DictConfig, OmegaConf, open_dict, read_write
 
+from hydra.core.config_loader import ConfigLoader
 from hydra.core.hydra_config import HydraConfig
 from hydra.core.singleton import Singleton
 from hydra.errors import HydraJobException
 from hydra.types import TaskFunction
 
+if TYPE_CHECKING:
+    from hydra._internal.callbacks import Callbacks
+
 log = logging.getLogger(__name__)
 
 
+@dataclass
+class HydraContext:
+    config_loader: ConfigLoader
+    callbacks: "Callbacks"
+
+
```

Comment:
hydra.types is potentially a better home for it (If placing it there does not introduce some hard circular dependencies).